modules/utils/2.0/utils.smk

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env snakemake
+
+
+##### ATTRIBUTION #####
+
+
+# Original Snakefile Author:    Helena Winata, Bruno Grande
+# Module Author:                Bruno Grande
+# Additional Contributors:      N/A
+
+
+##### SETUP #####
+
+import sys, os
+from os.path import join
+
+# Import package with useful functions for developing analysis modules
+import oncopipe as op
+
+_UTILS = config["lcr-modules"]["utils"]
+LOG = "/logs/" + op._session.launched_fmt
+
+wildcard_constraints:
```

Comment:
What does this constraint do?

Reply:
it makes sure that `out_dir` has the format `results/<module_name>-1.0` and `prefix` starts with the subdirectories `02-picard_qc/...`. This is done because `utils` doesn't rely on `op.setup_module` and so we can insert the `LOG` path to create the log directories.